### PR TITLE
Media foundation encoders

### PR DESCRIPTION
**Pull request by sethcarlton**

### CODE REVIEW

File: plugins/win-mf/mf-h264.cpp [hunk 1]
```diff
@@ -534,8 +534,8 @@ void RegisterMFH264Encoders()
 	for (auto e : encoders) {
 		/* ignore the software encoder due to the fact that we already
 		 * have an objectively superior software encoder available */
-		if (e->Type() == EncoderType::H264_SOFTWARE)
-			continue;
+		//if (e->Type() == EncoderType::H264_SOFTWARE)
+		//	continue;
```
Comment (cluangrath) [inline plugins/win-mf/mf-h264.cpp]: this is going into our own fork of OBS studio right? why not just delete the commented lines? keep the comment about the superior software encoder though lol

File: cmake/Modules/CopyMSVCBins.cmake [hunk 2]
```diff
@@ -53,41 +52,7 @@ file(GLOB FFMPEG_BIN_FILES
...
-	"${FFMPEG_avcodec_INCLUDE_DIR}/bin${_bin_suffix}/opus*.dll"
-	"${FFMPEG_avcodec_INCLUDE_DIR}/../bin/libbz2*.dll"
-	"${FFMPEG_avcodec_INCLUDE_DIR}/../bin/zlib*.dll"
-	"${FFMPEG_avcodec_INCLUDE_DIR}/bin/libbz2*.dll"
-	"${FFMPEG_avcodec_INCLUDE_DIR}/bin/zlib*.dll"
-
-	"${FFMPEG_avcodec_INCLUDE_DIR}/../bin${_bin_suffix}/libbz2*.dll"
-	"${FFMPEG_avcodec_INCLUDE_DIR}/../bin${_bin_suffix}/zlib*.dll"
-	"${FFMPEG_avcodec_INCLUDE_DIR}/bin${_bin_suffix}/libbz2*.dll"
-	"${FFMPEG_avcodec_INCLUDE_DIR}/bin${_bin_suffix}/zlib*.dll"
```
Comment (cluangrath) [inline cmake/Modules/CopyMSVCBins.cmake:89]: similar to the way the other non-ilos builds settings are being set, should we include this stuff if it's not an ilos build?

File: plugins/win-mf/mf-h264.cpp [hunk 3]
```diff
@@ -250,20 +250,20 @@ static void MFH264_GetDefaults(obs_data_t *settings)
 #define PROP_DEF(x, y, z) obs_data_set_default_ ## x(settings, y, z)
 	PROP_DEF(int,    MFP_BITRATE,         2500);
 	PROP_DEF(bool,   MFP_USE_LOWLAT,      true);
-	PROP_DEF(int,    MFP_B_FRAMES,        2);
+	PROP_DEF(int,    MFP_B_FRAMES,        0);
```
Comment (cluangrath) [inline plugins/win-mf/mf-h264.cpp:254]: yes yes, I know exactly what these values mean and where they come from

**Review by cluangrath — APPROVED** [review]

File: cmake/Modules/CopyMSVCBins.cmake (hunk 2 above)
Comment (sethcarlton) [reply cmake/Modules/CopyMSVCBins.cmake:89]: Could potentially do that. I think the ILOS_BUILD flag is relatively pointless as of now. It is probably better to just maintain our own branch and then if we are doing a non ilos build just use the obs branch

File: plugins/win-mf/mf-h264.cpp (hunk 3 above)
Comment (sethcarlton) [reply plugins/win-mf/mf-h264.cpp:254]: Just trust 😂

File: plugins/win-mf/mf-h264.cpp (hunk 1 above)
Comment (sethcarlton) [reply plugins/win-mf/mf-h264.cpp]: Yea I can just delete it at this point. It was just for ease of jumping back and forth

File: plugins/win-mf/mf-h264.cpp [hunk 4]
```diff
@@ -250,20 +250,20 @@ static void MFH264_GetDefaults(obs_data_t *settings)
...
 	PROP_DEF(bool,   MFP_USE_BUF_SIZE,    false);
 	PROP_DEF(int,    MFP_BUF_SIZE,        2500);
 	PROP_DEF(bool,   MFP_USE_MAX_BITRATE, false);
 	PROP_DEF(int,    MFP_MAX_BITRATE,     2500);
 	PROP_DEF(int,    MFP_KEY_INT,         2);
-	PROP_DEF(int,    MFP_RATE_CONTROL,    H264RateControlCBR);
+	PROP_DEF(int,    MFP_RATE_CONTROL,    H264RateControlVBR);
```
Comment (peterfares) [inline plugins/win-mf/mf-h264.cpp:260]: Question, why are we changing the defaults here vs passing in different options from our side?

Comment (sethcarlton) [reply plugins/win-mf/mf-h264.cpp:260]: Passing in options from our side doesn't actually update the media foundation encoder. It seems like they get set during initialization and then the update method doesn't update it appropriately. I am actually not sure if the current encoder updates work either. This basically was just a faster solution.

**Review by peterfares — APPROVED** [review]

**Pull request merged**